### PR TITLE
needed for IEX templates editing 

### PR DESCRIPTION
**Pull request by asasasasa**

### CODE REVIEW

File: src/service/InteractiveExposeService.ts [hunk 1]
```diff
@@ -1,190 +1,307 @@
...
-import { APIClient, APIMapping } from '../http';
+  DesignTemplate,
+  DesignTemplateAssignment,
+  InteractiveExposeMapping,
+  InteractiveExposeSettings,
+  InteractiveExposeSettingsWithLogos,
+  InteractiveExposeTemplate,
+  S3File,
+  SendInteractiveExposeModel
+} from "@flowfact/types";
```
Comment (ffjanhoeck) [inline src/service/InteractiveExposeService.ts]: Please use single quotes

File: src/service/InteractiveExposeService.ts [hunk 2]
```diff
@@ -1,190 +1,307 @@
...
 export class InteractiveExposeService extends APIClient {
+  constructor() {
+    super(APIMapping.interactiveExposeService);
+  }
 
-    constructor() {
-        super(APIMapping.interactiveExposeService);
-    }
+  async getSettings(): Promise<InteractiveExposeSettingsWithLogos> {
+    return (await this.invokeApi("/settings", "GET")).data;
```
Comment (ffjanhoeck) [inline src/service/InteractiveExposeService.ts]: Single quotes everywhere please

File: src/service/InteractiveExposeService.ts [hunk 3]
```diff
@@ -1,190 +1,307 @@
...
+    );
+  }
 
-    /**
-     * Perform the setup steps for the domain.
-     */
-    async setupDomain(): Promise<AxiosResponse> {
-        return this.invokeApi('/domain/setup', 'POST');
-    }
+  async getInteractiveExposeTemplateFiles(templateName: string) {
```
Comment (ffjanhoeck) [inline src/service/InteractiveExposeService.ts]:
```suggestion
  async fetchInteractiveExposeTemplateFiles(templateName: string) {
```

File: src/service/InteractiveExposeService.ts [hunk 4]
```diff
@@ -1,190 +1,307 @@
...
+    );
+  }
 
-    /**
-     * Get interactive expose mapping. Default mapping is just named so, cause of convienience
-     */
-    async getMappingOrDefault(schemaId: string = 'defaultMapping'): Promise<AxiosResponse> {
-        return this.invokeApi(`/mapping/${schemaId}`, 'GET');
-    }
+  async getInteractiveExposeTemplateFilesByFileType(
```
Comment (ffjanhoeck) [inline src/service/InteractiveExposeService.ts]:
```suggestion
  async fetchInteractiveExposeTemplateFilesByFileType(
```

File: src/service/InteractiveExposeService.ts [hunk 5]
```diff
@@ -1,190 +1,307 @@
...
+    );
+  }
 
-    /**
-     * Get interactive expose mapping. Default mapping is just named so, cause of convienience
-     */
-    async getAllMappings(): Promise<AxiosResponse> {
-        return this.invokeApi('/mapping/all', 'GET');
-    }
+  async getInteractiveExposeTemplateFileData(
```
Comment (ffjanhoeck) [inline src/service/InteractiveExposeService.ts]:
```suggestion
  async fetchInteractiveExposeTemplateFileData(
```

File: src/service/InteractiveExposeService.ts [hunk 6]
```diff
@@ -1,190 +1,307 @@
...
+    templateName: string,
+    fileType: string,
+    body: string[]
+  ) {
+    return this.invokeApi(
+      "/public/interactiveExposes/templates/" +
```
Comment (ffjanhoeck) [inline src/service/InteractiveExposeService.ts]: Please use ES6 String templates instead
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/template_strings

File: src/service/InteractiveExposeService.ts (hunk 1 above)
Comment (asasasasa) [reply src/service/InteractiveExposeService.ts]: Will do replace all, and before was "

File: src/service/InteractiveExposeService.ts (hunk 2 above)
Comment (asasasasa) [reply src/service/InteractiveExposeService.ts]: Will do replace all, and before was "

File: src/service/InteractiveExposeService.ts (hunk 6 above)
Comment (asasasasa) [reply src/service/InteractiveExposeService.ts]: done

File: src/service/InteractiveExposeService.ts [hunk 7]
```diff
@@ -1,190 +1,300 @@
...
+    );
+  }
 
-    /**
-     * Add a domain to the company.
-     */
-    async addDomain(domain: string): Promise<AxiosResponse> {
-        return this.invokeApi('/domain', 'POST', '', { queryParams: { domain } });
-    }
+  async uploudInteractiveExposeTemplateFiles(
```
Comment (ffjanhoeck) [inline src/service/InteractiveExposeService.ts]:
```suggestion
  async uploadInteractiveExposeTemplateFiles(
```

**Pull request merged**